cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -80,6 +84,25 @@ struct TemporalComponentExtract {
   }
 };
 
+template <typename Op, typename OutType>
+struct DayOfWeekOptionsWithOptions {
```

Comment:
This name is... weird. Can you find something better?

Reply:
Indeed. `DayOfWeekWithOptions`?

Reply:
Well, it's still confusing. `DayOfWeekExec`?

Reply:
Sure.